### PR TITLE
AC: write result in csv

### PR DESCRIPTION
**Pull request by eaidova**
simple accuracy checker results writing. Can be useful for manual testing

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 1]
```diff
@@ -266,5 +277,32 @@ def print_processing_info(model, launcher, device, tags, dataset):
...
+            'metric_name': metric_result['name'],
+            'metric_type': metric_result['type'],
+            'metric_value': metric_result['value']
+        })
+        rows.append(result)
+    with open(csv_file, 'a+', newline='') as f:
+        writer = DictWriter(f, fieldnames=field_names)
+        if new_file:
+            writer.writeheader()
+        writer.writerows(rows)
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/main.py]: Is it really necessary to accumulate the rows in memory first? You could just move the loop here, and write them into the file one by one.

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 2]
```diff
@@ -266,5 +277,32 @@ def print_processing_info(model, launcher, device, tags, dataset):
...
+        'dataset': dataset
+    }
+    rows = []
+    for metric_result in metric_results:
+        result = deepcopy(main_info)
+        result.update({
+            'metric_name': metric_result['name'],
+            'metric_type': metric_result['type'],
+            'metric_value': metric_result['value']
+        })
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/main.py]: `deepcopy` seems excessive, why not just `result = {**main_info, metric_name: metric_result['name'], ...}`?

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/main.py]: fixed

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 1 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/main.py]: fixed

**Pull request merged**